cache/models/samples.js

```diff
@@ -393,10 +395,11 @@ function handleUpsertError(objectType, isBulk) {
  * else creates a new one.
  * @param  {Object} sampleQueryBodyObj - Query Body Object for a sample
  * @param  {Boolean} isBulk - If the caller method is bulk upsert
- * @param {String} userName - The user performing the write operation
+ * @param {Object} user - The user performing the write operation
  * @returns {Object} - Updated sample
  */
-function upsertOneSample(sampleQueryBodyObj, isBulk, userName) {
+function upsertOneSample(sampleQueryBodyObj, isBulk, user) {
+  const userName = user ? user.name : false;
```

Comment:
@annyhe Any reason why you are using "false" instead of "undefined" here and a few other places?